src/emcpy/utils/dateutils.py

```diff
@@ -0,0 +1,226 @@
+    Args:
+        yyyy: (int) year
+        mm: (int) month
+        dd: (int) day
+        hh: (int) hour
+    Returns:
+        yyyymmddhh: (date str) date
+    """
+    return "%0.4i" % (yyyy) + "%0.2i" % (mm) + "%0.2i" % (dd) + "%0.2i" % (hh)
+
+
+def hrstodate(hrs, mixedcal=True):
+    """
+    Args:
+        hrs:
+        mixedcal: (bool) is mixed Gregorian/Julian calendar used?
+    Returns:
+        date string of the form yyyymmddhh given hrs since day 1 CE.
+    """
+    date = hrs_since_day1CE_todate(hrs, mixedcal=mixedcal)
+    return makedate(date.year, date.month, date.day, date.hour)
+
+
+def datetohrs(yyyymmddhh, mixedcal=True):
+    """
+    Args:
+        yyyymmddhh: (date str) date
+        mixedcal: (bool) is mixed Gregorian/Julian calendar used?
+    Returns:
+        hours since day 1 CE given a date string of the form yyyymmddhh
+    """
+    yyyy, mm, dd, hh = splitdate(yyyymmddhh)
+    return dateto_hrs_since_day1CE(datetime.datetime(yyyy, mm, dd, hh), mixedcal=mixedcal)
+
+
+def daterange(date1, date2, hrinc):
+    """
+    return a list of date strings of the form yyyymmddhh given
+    a starting date, ending date and an increment in hours.
+    Args:
+        date1: (date str) start date
+        date2: (date str) end date
+        hrinc: (int) increment in hours
+    Returns:
+        dates: (list date str) of the form yyyymmddhh
+    """
+    date = date1
+    delta = datetime.timedelta(hours=1)
+    yyyy, mm, dd, hh = splitdate(date)
+    d = datetime.datetime(yyyy, mm, dd, hh)
+    n = 0
+    dates = [date]
+    while date < date2:
+        d = d + hrinc * delta
+        date = makedate(d.year, d.month, d.day, d.hour)
+        dates.append(date)
+        n = n + 1
+    return dates
+
+
+def dayofyear(yyyy, mm, dd):
+    """
+    Args:
+        yyyy: (int) year
+        mm: (int) month
+        dd: (int) day
+        hh: (int) hour
+    Returns:
+        integer day of year given yyyy,mm,dd
+    """
+    d = datetime.datetime(yyyy, mm, dd)
+    d0 = datetime.datetime(yyyy, 1, 1)
+    return (d - d0).days
+
+
+def getyrmon(day_of_year, yyyy=2001):
+    """
+    Args:
+        day_of_year: (int) day of the year
+        yyyy: (int) year
+    Returns:
+        mm: (int) month of year given day of year
+        dd: (int) day of month given day of year
+    """
+    d1 = datetime.datetime(yyyy, 1, 1)
+    if calendar.isleap(d1.year) and day_of_year > 366:
+        raise ValueError("not that many days in the year")
+    if not calendar.isleap(d1.year) and day_of_year > 365:
+        raise ValueError("not that many days in the year")
+    d2 = d1 + (day_of_year - 1) * datetime.timedelta(days=1)
+    return d2.month, d2.day
+
+
+def daysinmonth(yyyy, mm):
+    """
+    Args:
+        yyyy: (int) year
+        mm: (int) month
+    Returns:
+        number of days in month given yyyy,mm
+    """
+    return calendar.monthrange(yyyy, mm)[1]
+
+
+if __name__ == "__main__":
```

Comment:
why is this needed?

Reply:
It would be useful for leap years and just general date calculations which might otherwise require hard coding of number of days in a month, for example.

Reply:
Sorry I mean the `__main__` part